src/txgraph.cpp

```diff
@@ -1191,25 +1186,17 @@ void GenericClusterImpl::ApplyRemovals(TxGraphImpl& graph, int level, std::span<
         todo.Reset(m_linearization.back());
         m_linearization.pop_back();
     }
-    if (todo.None()) {
-        // If no further removals remain, and thus all removals were at the end, we may be able
-        // to leave the cluster at a better quality level.
-        if (IsAcceptable(/*after_split=*/true)) {
-            quality = QualityLevel::NEEDS_SPLIT_ACCEPTABLE;
-        } else {
-            quality = QualityLevel::NEEDS_SPLIT;
-        }
-    } else {
+    if (todo.Any()) {
```

Comment:
1b2c53ae0c9020791d39c85b032f7f907b468c99

Could we go another step further and delete the 

>// First remove all removals at the end of the linearization

Stuff, drop this specific conditional and just filter or is this still a performance win